### ISSUE TITLE
PVC spec inconsistent with k8s

### ISSUE THREAD

**Issue by pkliczewski** [kubevirt/kubevirt#3177]
We see that our kubevirt java [client](https://github.com/oVirt/java-client-kubevirt) generated from kubevirt swagger file [fails](https://bugzilla.redhat.com/1814215) to unmarshall virtual machines with data volume defined. I investigated the issue and it seems there is a difference in `PersistentVolumeClaimSpec` definition.

In [k8s](https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json) we have:

```
    "io.k8s.api.core.v1.PersistentVolumeClaimSpec": {
      "description": "PersistentVolumeClaimSpec describes the common attributes of storage devices and allows a Source for provider-specific attributes",
      "properties": {
        "accessModes": {
          "description": "AccessModes contains the desired access modes the volume should have. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1",
          "items": {
            "type": "string"
          },
          "type": "array"
        },
```

and in [Kubevirt](https://github.com/kubevirt/kubevirt/blob/master/api/openapi-spec/swagger.json):

```
   "v1.PersistentVolumeClaimSpec": {
    "description": "PersistentVolumeClaimSpec describes the common attributes of storage devices and allows a Source for provider-specific attributes",
    "properties": {
     "accessModes": {
      "description": "AccessModes contains the desired access modes the volume should have. More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#access-modes-1",
      "type": "array",
      "items": {
       "$ref": "#/definitions/v1.PersistentVolumeAccessMode"
      }
     },
```

Please make the api consistent so we can regenerate our client. @mhenriks reported there was similar issue open in CDI. https://github.com/kubevirt/containerized-data-importer/issues/1084

**Comment by pkliczewski** [comment]
We found exactly the same issue for 2 more types `EvictionStrategy` and `VolumeMode`. Please check whether there are more types defined in this way.

**Comment by pkliczewski** [comment]
I went over swagger file and I see that there are more types defined like this. It seems like we are generating the file using buggy or different tools than k8s. I checked [PodQOSClass](https://github.com/kubernetes/kubernetes/blob/0fd4beb089a31642d7f86bb561b66568ab4afa59/pkg/apis/core/types.go#L2898) and it is defined as string in k8s api [definition](https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json):

```
        "qosClass": {
          "description": "The Quality of Service (QOS) classification assigned to the pod based on resource requirements See PodQOSClass type for available QOS classes More info: https://git.k8s.io/community/contributors/design-proposals/node/resource-qos.md",
          "type": "string"
        },
```
 but in kubevirt we have [PodQOSClass type](https://github.com/kubevirt/kubevirt/blob/master/api/openapi-spec/swagger.json#L5810) instead of a string.

**Comment by pkliczewski** [comment]
@rmohr @fabiand fyi

**Comment by rmohr** [comment]
As discussed offline, @pkliczewski it would be great if you switch kubevirt to use the k8s api generator when you find some time.

**Comment by rmohr** [comment]
Fixed in #3357. See https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachineinstancestatus.

**Comment by rmohr** [comment]
Will be part of 0.30.0.

/close
